src/fingerprint.cpp

```diff
@@ -27,12 +27,33 @@ std::string uri_fingerprint(const std::string& uri) {
 
     std::stringstream fingerprint;
 
-    Poco::URI uri_parsed(uri);
-    const std::string& path = uri_parsed.getPath();
+    // faup_options_new() is not thread safe, and should only be runned once per
+    // code, it is also the part that loads the cached publicsuffix.org file
+    faup_options_t* faup_opts;
```

Comment:
We should investigate to see if this pointer needs to be freed
Even tho I haven't found any usage of freeing method in the [faup documentation](https://github.com/stricaud/faup/blob/master/doc/library.md) I found [this example using `faup_options_free(faup_opts);`](https://cpp.hotexamples.com/fr/examples/-/-/faup_init/cpp-faup_init-function-examples.html)